assets/app/scripts/directives/nav.js

```diff
@@ -53,11 +53,16 @@ angular.module('openshiftConsole')
 
           sortedProjects = $filter('orderByDisplayName')(projects);
 
-          options = _.map(sortedProjects, function(item) {
+          options = _.map(sortedProjects, function(item, index) {
```

Comment:
Service seems heavyweight. Filter that takes the item and the list (e.g. `uniqueDisplayName(item,items)`) would be much cleaner, though it would have to calculate all display names every time